Documentações de Setup e Contribuição

PedroAfb opened the pull request:
Este pull request melhora a documentação do projeto ao introduzir diretrizes de contribuição mais claras e separar as instruções de configuração em um documento dedicado. As principais mudanças têm como objetivo facilitar para os contribuidores começarem e entenderem o processo de contribuição.

Melhorias na documentação:

* Adicionado um novo arquivo `CONTRIBUTING.md` com instruções detalhadas sobre os tipos de contribuições aceitas, fluxo de trabalho e boas práticas para o projeto.

* Criado um novo arquivo `setup.md` com instruções passo a passo para configurar tanto o frontend quanto o backend localmente, incluindo pré-requisitos, instalação de dependências, configuração de ambiente e execução da aplicação.

Reorganização da documentação:

* Atualizado o `README.md` para remover as instruções de configuração inline e, em vez disso, adicionar um link para o novo setup.md como guia de configuração.

PedroAfb committed:
como rodar o projeto localmente

PedroAfb committed:
docs de contribuição

PedroAfb committed:
atualizando links das docs

PedroAfb commented:
Algumas dúvidas:

* O repositório do backend é privado, então a seção do `zerobito-backend` no arquivo [setup.md](https://github.com/SocialSoftwareLivingLab/zerobito/blob/feat/docs/setup.md) seria útil apenas para quem tem acesso a ele. Nesse caso, quem não tiver acesso não conseguiria testar a aplicação. Gostaria de saber como vocês acham que devemos proceder nesse ponto.

* Além disso, gostaria de saber a opinião de vocês sobre o restante das alterações propostas neste PR.

brenofranca83 commented:
> Algumas dúvidas:
> 
> * O repositório do backend é privado, então a seção do `zerobito-backend` no arquivo [setup.md](https://github.com/SocialSoftwareLivingLab/zerobito/blob/feat/docs/setup.md) seria útil apenas para quem tem acesso a ele. Nesse caso, quem não tiver acesso não conseguiria testar a aplicação. Gostaria de saber como vocês acham que devemos proceder nesse ponto.
> * Além disso, gostaria de saber a opinião de vocês sobre o restante das alterações propostas neste PR.

Valeu por observar isso, Pedro. O repo do backend agora está público.
Estruturalmente, eu gostei da forma que está. Mais detalhes sobre o PR eu vou fazer na revisão dele.

brenofranca83 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+
+
+5. **Pull Requests** 
+   * contra a branch `develop` do repositório principal.
```
Devemos remover a branch develop no curto prazo, então a ideia é atualiza isso para a 'main'

brenofranca83 commented on setup.md:
```diff
@@ -0,0 +1,57 @@
+## Requisitos Prévios
+
+Antes de iniciar, certifique-se de ter instalado em seu sistema:
+
```
é bom ter as versões mínimas desses requisitos

brenofranca83 commented on setup.md:
```diff
@@ -0,0 +1,57 @@
+
+
+## Repositório `zerobito` - Aplicação Frontend
+- Rode o comando `yarn` para instalar as dependências e o script `yarn start` para subir a aplicação
```
O primeiro comando é só 'yarn'?
Não seria yarn build ou algo assim?

brenofranca83 commented on setup.md:
````diff
@@ -0,0 +1,57 @@
+    ```bash
+    yarn start:dev
+    ```
+  - ℹ️ Na primeira execução, o servidor cria toda a base de dados. Após isso, pare o servidor e rode novamente para que ele consiga inicializar corretamente.
````
Nossa, isso está ruim. Ele já deveria inicializar corretamente.

brenofranca83 commented on setup.md:
````diff
@@ -0,0 +1,57 @@
+    yarn start:dev
+    ```
+  - ℹ️ Na primeira execução, o servidor cria toda a base de dados. Após isso, pare o servidor e rode novamente para que ele consiga inicializar corretamente.
+- Há um usuário root default, criado pelo seed [usuario-administrador.seed.ts](https://github.com/SocialSoftwareLivingLab/zerobito-backend/blob/develop/src/app/usuarios/seeds/usuario-administrador.seed.ts).
````
precisa dizer a senha ou para alterar a senha, não?

brenofranca83 commented on setup.md:
```diff
@@ -0,0 +1,57 @@
+  - ℹ️ Na primeira execução, o servidor cria toda a base de dados. Após isso, pare o servidor e rode novamente para que ele consiga inicializar corretamente.
+- Há um usuário root default, criado pelo seed [usuario-administrador.seed.ts](https://github.com/SocialSoftwareLivingLab/zerobito-backend/blob/develop/src/app/usuarios/seeds/usuario-administrador.seed.ts).
+
+**Agora a aplicação está pronta para teste em [http:localhost:3000/](http://localhost:3000/)**
```
Esse endereço é do front, não?

PedroAfb committed:
PR deve ser para a branch main

PedroAfb committed:
versões mínimas para os requisitos

PedroAfb committed:
alterações setup.md
deixando mais claro a disponibilidade inicial de um login de user root
corrigindo a informações sobre o endereço do front e back
